### DISCUSSION TITLE
Error: operation has timed out

### DISCUSSION THREAD

**Discussion by rasheed-asahi** [socketio/socket.io#4568] [Q&A]
Frequently getting this error "Error: operation has timed out" in my sails logs. The error message points to this location "socket.io\dist\broadcast-operator.js:181:30".

Can somebody help me with this?

Error from sails log given below: 
Error: operation has timed out
    at Timeout._onTimeout (socket.io/dist/broadcast-operator.js:181:30)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7)

Node Version:16.10.0
Socket.io Version: 4.5.4
Sails Version: 1.5.3

**Comment by darrachequesne** [comment]
`operation has timed out` can be returned when sending a packet to multiple clients and expecting an acknowledgement ([here](https://github.com/socketio/socket.io/blob/main/lib/broadcast-operator.ts#L233) in the code), but not all clients respond in the given delay:

```js
io.timeout(5000).emit("hello", "world", (err, responses) => {
  if (err) {
    // some clients did not acknowledge the event in the given delay
  } else {
    console.log(responses); // one response per client
  }
});
```

Reference: https://socket.io/docs/v4/broadcasting-events/#with-acknowledgements

**Reply by ghost** [reply]
Hi @darrachequesne, the `timeout()` call is mandatory even when calling `socket.emit()` on a remote fetched socket emitting only to that single specific socket? Because I see the same error in that case... Is considered broadcasting that too?

**Reply by ghost** [reply]
I think this may be related: https://github.com/socketio/socket.io/pull/4558

**Reply by darrachequesne** [reply]
> the `timeout()` call is mandatory even when calling `socket.emit()` on a remote fetched socket emitting only to that single specific socket? Because I see the same error in that case... Is considered broadcasting that too?

Yes it is. I will make it clearer in the documentation.

**Comment by potados99** [comment]
I added `.timeout(1000)` and solved.
